changelogs/fragments/85623-fix-pip-check-mode-vcs.yml

```
@@ -0,0 +1,3 @@
+bugfixes:
+  - ``ansible.builtin.pip`` - Running the built-in pip module with ``check_mode`` and packages coming from VCS URLs, now correctly outputs sets the ``changed`` status of the task. Previously, it was always reported as changed due to improper package name resolution
+    (https://github.com/ansible/ansible/pull/85623)
```

Looks like YAML is sensitive to leading backticks:
```suggestion
  - >-
    ``ansible.builtin.pip`` - Running the built-in pip module with ``check_mode`` and packages coming from VCS URLs, now correctly outputs sets the ``changed`` status of the task.
    Previously, it was always reported as changed due to improper package name resolution
    (https://github.com/ansible/ansible/pull/85623)
```